src/widgetresize/resizerstate.js

```diff
@@ -146,6 +141,23 @@ export default class ResizeState {
 		this.proposedHandleHostWidth = newSize.handleHostWidth;
 		this.proposedHandleHostHeight = newSize.handleHostHeight;
 	}
+
+	/**
+	 * Calculates a relative width of a `domResizeHost` compared to it's parent in percents.
+	 *
+	 * @protected
+	 * @param {HTMLElement} domResizeHost
+	 * @returns {Number}
+	 */
+	_getParentPercentage( domResizeHost ) {
+		const rect = new Rect( domResizeHost );
+		const domResizeHostParent = domResizeHost.parentElement;
```

Comment:
Isn't this a bit naive? The image still blinks heavily on the first resize event (which I know is a separate issue), so it's a bit hard to tell how well we calculate the width. But if you're also aware that this is a naive solution, then it'd be good to comment this in the code.

Reply:
> The image still blinks heavily on the first resize event (which I know is a separate issue), so it's a bit hard to tell how well we calculate the width.

I'm pretty confident that this fix resolves the problem of this issue. You can see that the values are now calculated correctly. As you mentioned, the only problem is this full-width flash which will be fixed in a separated issue.

> Isn't this a bit naive? The image still blinks heavily on the first resize event (which I know is a separate issue), so it's a bit hard to tell how well we calculate the width.

TBH, I don't see anything too naive here. Max element's width is simply constrained by the parent, which is a `div`, `td`, `th` or `blockquote` etc.

If we want to make it more complex, then we can go with [`HTMLElement.offsetParent`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent).

The benefit of this solution is that it will work well with an image wrapped in a transparent content model container &mdash; I'm thinking mostly about `<a>` element, if we ever allow to wrap multiple blocks in a single `<a>` element. Without this, the current proposal would use this `<a>` directly and use `auto` as a width value, which will cause an issue.

The problem with `offsetParent` is that it comes with more cases where a `null` is returned and it would be nice to handle it somewhat gracefully. This would require `elem.offsetParent || <fallbackElement>` here, which would in turn require us to write weird and unrealistic test cases for code coverage.

All in all, simple parent usage works well for all the cases that we have today, so I see no reason to further complicate it for the time being.